src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+using System.Security;
+using Internal.NativeFormat;
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+using System.Text;
+using System;
+using System.Runtime;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    internal static class StartupCodeHelpers
+    {
+        internal static string[] CommandLineArgs { get; set; }
```

Comment:
It may be better for the dependency to go other way - the System.Environment should have the static variable, and it should be set from here.